sub-packages/bionemo-esm2/src/bionemo/esm2/run/recipes.py

```diff
@@ -0,0 +1,447 @@
+        save_top_k=2,
+        create_tensorboard_logger=False,
+    )
+
+
+def esm2_tiny_model_config(
+    seq_length: int = 2048,
+    precision: PrecisionTypes = "bf16-mixed",
+    nemo1_init_path: Optional[str] = None,
+    initial_ckpt_path: Optional[str] = None,
+    biobert_spec_option: BiobertSpecOption = BiobertSpecOption.esm2_bert_layer_with_transformer_engine_spec,
+    variable_seq_lengths: bool = False,
+) -> ExposedESM2PretrainConfig:
+    """Model config for ESM2 tiny, used for testing."""
+    return ExposedESM2PretrainConfig(
+        seq_length=seq_length,
+        num_layers=2,
+        hidden_size=32,
+        num_attention_heads=2,
+        ffn_hidden_size=4 * 32,
+        params_dtype=precision,
+        pipeline_dtype=precision,
+        autocast_dtype=precision,
+        biobert_spec_option=biobert_spec_option,
+        nemo1_ckpt_path=str(nemo1_init_path) if nemo1_init_path is not None else None,
+        # handle checkpoint resumption here rather than auto-resume so this supports fine-tuning capabilities
+        initial_ckpt_path=str(initial_ckpt_path) if initial_ckpt_path is not None else None,
+        variable_seq_lengths=variable_seq_lengths,
+    )
+
+
+def esm2_tiny_test_recipe(args):
+    """Test recipe for ESM2 tiny, used for testing."""
+    parallel_config = simple_parallel_recipe()
+    training_config = tiny_train_config_recipe()
+
+    data_config = ESM2DataConfig(
+        min_seq_length=128,
+        max_seq_length=128,
+        micro_batch_size=2,
+        num_dataset_workers=1,
+        train_cluster_path=args.train_cluster_path,
+        train_database_path=args.train_database_path,
+        valid_cluster_path=args.valid_cluster_path,
+        valid_database_path=args.valid_database_path,
+    )
+    bionemo_model_config = esm2_tiny_model_config(
+        seq_length=data_config.max_seq_length, initial_ckpt_path=args.initial_ckpt_path
+    )
+
+    optim_config = default_adam_optimizer_with_cosine_annealing_recipe()
+    experiment_config = experiment_config_recipe(args.result_dir)
+    wandb_config = WandbConfig(
+        project="bionemo2-demo",
+        entity="nvidia",
+        offline=True,
+        tags=[],
+        group="dev",
+        id="dev",
+        log_model=False,
+        anonymous=True,
+    )
+    main_config = MainConfig[ExposedESM2PretrainConfig, ESM2DataConfig](
+        data_config=data_config,
+        parallel_config=parallel_config,
+        training_config=training_config,
+        bionemo_model_config=bionemo_model_config,
+        optim_config=optim_config,
+        experiment_config=experiment_config,
+        wandb_config=wandb_config,
+    )
+    return main_config
+
+
+def main():  # noqa: D103
+    def parse_args():
+        parser = argparse.ArgumentParser(description="Create ESM2 configuration JSON.")
+        parser.add_argument(
+            "--recipe",
+            type=str,
+            choices=["test", "8m", "650m", "3b"],
```

Comment:
We should include 15B.

Reply:
Do we ship the 15B checkpoint?

Reply:
@farhadrgh indicated we are only shippint 3b, let's do this when we have a checkpoint to ship (or next release)